### PR TITLE
Update minimum AIX pinnings to match non AIX builds

### LINKED ISSUE #12940: Should the minimum numpy for AIX be bumped to 1.16.5
https://github.com/scipy/scipy/blob/37b650e04f3bf49bc11cfcd18f2848ad4f957a0d/pyproject.toml#L9

I'm trying to update the minimum numpy for scikit-image, and noticed that you have different numpy minor versions for AIX and not AIX.

Is there a reason for this?

<details>
Historic conversation can be found here:
https://github.com/scipy/scipy/commit/46df8c0013e97a025376f1bd0ef07bbd8372aca7
</details>

### PR DESCRIPTION
**Pull request by hmaarrfk**
#### Reference issue
Fixes #12940

#### What does this implement/fix?
The AIX pinning is lower than the non AIX. When raised in #12940, @rgommers indicated that this might be an oversight since there are no CI builds to catch these kinds of errors.
https://github.com/scipy/scipy/issues/12940#issuecomment-707181364

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -6,8 +6,8 @@ requires = [
     "numpy==1.16.5; python_version=='3.6' and platform_system!='AIX'",
     "numpy==1.16.5; python_version=='3.7' and platform_system!='AIX'",
     "numpy==1.17.3; python_version=='3.8' and platform_system!='AIX'",
-    "numpy==1.16.0; python_version=='3.6' and platform_system=='AIX'",
-    "numpy==1.16.0; python_version=='3.7' and platform_system=='AIX'",
+    "numpy==1.16.5; python_version=='3.6' and platform_system=='AIX'",
+    "numpy==1.16.5; python_version=='3.7' and platform_system=='AIX'",
     "numpy==1.17.3; python_version=='3.8' and platform_system=='AIX'",
```
Comment (peterbell10) [inline pyproject.toml:11]: Since they all match now, is there any reason not to drop the `platform_system` clause entirely?

Comment (hmaarrfk) [reply pyproject.toml:11]: thats fair, but keeping them different helps remember hte history a little bit.

Comment (peterbell10) [reply pyproject.toml]: I sort of see your point, but having it in two places is why these didn't get updated in the first place. Better to have a "single source of truth".

Comment (hmaarrfk) [reply pyproject.toml]: you are right. the history isn't even that well conveyed in the code. So there isnt' really any advantage.

**Review by rgommers — APPROVED** [review]
Thanks @hmaarrfk

**Pull request merged**